provider/validators/cdo_role.go

```diff
@@ -0,0 +1,102 @@
+package validators
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/golang-jwt/jwt/v5"
+	"github.com/hashicorp/terraform-plugin-framework-validators/helpers/validatordiag"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+const (
+	ErrInvalidTokenFormat = "invalid JWT token format"
+	ErrDecodeFailed       = "failed to decode token payload"
+	ErrNoRolesFound       = "no roles found in the JWT token"
+)
+
+var _ validator.String = oneOfRolesValidator{}
+
+// oneOfRolesValidator validates that the value matches one of expected roles.
+type oneOfRolesValidator struct {
+	expectedRoles []types.String
+}
+
+func (v oneOfRolesValidator) Description(ctx context.Context) string {
+	return v.MarkdownDescription(ctx)
+}
+
+func (v oneOfRolesValidator) MarkdownDescription(_ context.Context) string {
+	return fmt.Sprintf("must contains one of CDO roles: %q", v.expectedRoles)
+}
+
+func (v oneOfRolesValidator) ValidateString(ctx context.Context, request validator.StringRequest, response *validator.StringResponse) {
+	if request.ConfigValue.IsNull() || request.ConfigValue.IsUnknown() {
+		return
+	}
+
+	token := request.ConfigValue.String()
+
+	role, err := extractRoleFromToken(token)
+
+	if err != nil {
+		fmt.Println("Error:", err)
+	}
+
+	for _, expectedRole := range v.expectedRoles {
+		if role == expectedRole.ValueString() {
+			return
+		}
+	}
+
+	response.Diagnostics.Append(validatordiag.InvalidAttributeValueMatchDiagnostic(
+		request.Path,
+		v.Description(ctx),
+		role,
+	))
+}
+
+func extractRoleFromToken(tokenString string) (string, error) {
```

Comment:
Shouldn't we do a nil check here? Or will that be caught earlier?

Reply:
sure will add, thanks